testbed/tests/app/test_desktop.py

```diff
@@ -168,159 +170,183 @@ async def test_menu_minimize(app, app_probe):
     assert window1_probe.is_minimized
 
 
-async def test_full_screen(app, app_probe):
```

Comment:
You need to be *very* careful here - you shouldn't ever *delete* a test, as it encodes behaviour that needs to be maintained. You might update an API if the old API has been deprecated, but you don't *delete* the test unless the behaviour being tested has *deliberately* changed.

In this case, the existing "test_full_screen" validates a bunch of transitions that the new tests don't validate.

Reply:
It's not about restoring the *literal* test - it's about restoring the *sprit* of the test. Although the public API has changed, the set of behaviours being tested hasn't. There's a very specific set of "window 1 full screen, direct to window 2 full screen" behaviours that this test encodes, but your replacement test doesn't. It's important that in updating the tests, we don't remove regression tests that were added in the past to verify specific behaviours.

Reply:
In the previous test `tests/app/test_desktop.py::test_full_screen`, there are 2 major things being tested:

**1.** When one window is in PRESENTATION(app full screen), then on entering PRESENTATION with another window without exiting PRESENTATION mode first, should make the first window exit PRESENTATION and only make the second window in PRESENTATION.
**2.** The increase in size of content and flex widgets, when entering presentation mode.

For testing the behavior of **1**, I have added a new test: `tests/app/test_desktop.py::test_window_presentation_exit_on_another_window_presentation`

Behavior of **2** is already being tested in `tests/app/test_desktop.py::test_presentation_mode`